dspace-iiif/src/main/java/org/dspace/app/iiif/service/ManifestService.java

```diff
@@ -102,17 +111,57 @@ public ManifestService(ConfigurationService configurationService) {
 
     /**
      * Returns JSON manifest response for a DSpace item.
+     * 
+     * Will retrieve the item's manifest bundle or generate a "default" manifest
+     * if the bundle isn't found.
      *
      * @param item the DSpace Item
      * @param context the DSpace context
      * @return manifest as JSON
      */
-    public String getManifest(Item item, Context context) {
+    public String getManifest(Item item, Context context) throws AuthorizeException, IOException, SQLException {
+        Bitstream manifestBitstream = getPrimaryOrFirstBitstream(item, Constants.IIIF_MANIFEST_BUNDLE_NAME);
+
+        if (manifestBitstream == null) {
+            return createManifest(item, context);
+        }
+
+        try (InputStream inputStream = bitstreamService.retrieve(context, manifestBitstream)) {
+            return new String(inputStream.readAllBytes(), StandardCharsets.UTF_8);
+        }
+    }
+
+    private Bitstream getPrimaryOrFirstBitstream(Item item, String bundleName) throws SQLException {
+        List<Bundle> bundles = itemService.getBundles(item, bundleName);
+
+        if (bundles == null || bundles.isEmpty()) {
+            return null;
+        }
+
+        Bundle bundle = bundles.get(0);
+        Bitstream result = bundle.getPrimaryBitstream();
+
+        if (result == null && !bundle.getBitstreams().isEmpty()) {
+            result = bundle.getBitstreams().get(0);
+        }
+
```

Comment:
It would be nice to validate the response in some way (if this hasn't already been done during upload). We can of course check that it's json. I wonder about also checking for a few properties in the response, like `@context`, `@type` and `@id`. The validation keys and maybe also their expected values could be configured in `iiif.cfg`

Reply:
In your IT file you ask whether we need to check for both the primary bitstream and the first bitstream. It's a good question. My guess is that the primary bitstream would not be used much and there should only be a single manifest file in almost all cases.